### ISSUE TITLE
Runtime of noise modeling depends on number of qubits in backend NOT the circuit

### ISSUE THREAD

**Issue by nonhermitian** [Qiskit/qiskit-aer#1087]
### Informations

- **Qiskit Aer version**: 0.7.2
- **Python version**:
- **Operating system**: OSX

### What is the current behavior?
The runtime of noise modeling for a given circuit depends on that number of qubits a system has, even though the circuit size is the same.

### Steps to reproduce the problem
Consider the following circuit:

```python
import numpy as np
from qiskit import *
from qiskit.circuit.library import EfficientSU2, TwoLocal

num_qubits = 4
qc = TwoLocal(num_qubits, ['h', 'rx'], 'cz', entanglement='linear', reps=2, parameter_prefix = 'y')
qc.draw()
```

```
     ┌───┐┌──────────┐   ┌───┐┌──────────┐                        ┌───┐»
q_0: ┤ H ├┤ RX(y[0]) ├─■─┤ H ├┤ RX(y[4]) ├─────────────────■──────┤ H ├»
     ├───┤├──────────┤ │ └───┘└──┬───┬───┘┌──────────┐     │      └───┘»
q_1: ┤ H ├┤ RX(y[1]) ├─■───■─────┤ H ├────┤ RX(y[5]) ├─────■────────■──»
     ├───┤├──────────┤     │     └───┘    └──┬───┬───┘┌──────────┐  │  »
q_2: ┤ H ├┤ RX(y[2]) ├─────■───────■─────────┤ H ├────┤ RX(y[6]) ├──■──»
     ├───┤├──────────┤             │         ├───┤    ├──────────┤     »
q_3: ┤ H ├┤ RX(y[3]) ├─────────────■─────────┤ H ├────┤ RX(y[7]) ├─────»
     └───┘└──────────┘                       └───┘    └──────────┘     »
«     ┌──────────┐                         
«q_0: ┤ RX(y[8]) ├─────────────────────────
«     └──┬───┬───┘┌──────────┐             
«q_1: ───┤ H ├────┤ RX(y[9]) ├─────────────
«        └───┘    └──┬───┬───┘┌───────────┐
«q_2: ─────■─────────┤ H ├────┤ RX(y[10]) ├
«          │         ├───┤    ├───────────┤
«q_3: ─────■─────────┤ H ├────┤ RX(y[11]) ├
«                    └───┘    └───────────┘
```
When doing noise modeling of the above circuit across multiple system of various sizes, the runtime increases more or less linearly with the number of qubits in the system.  This was measured using `job.result().metadata['time_taken']`, and averaged over 10 instances of 8192 shots.  This despite the fact that the circuit is the exact same in all the cases (layout was manually fixed to make sure this is the case).

<img width="496" alt="Screen Shot 2020-12-24 at 06 45 50" src="https://user-images.githubusercontent.com/1249193/103086851-aedb4400-45b3-11eb-9b5d-59d572e3f47a.png">

**Comment by chriseclectic** [comment]
If layout is being specified so that circuit truncation isn't needed I would guess this is likely due to the time taken to deserialize the noise model from JSON in the in the C++ code.

**Comment by nonhermitian** [comment]
Hmm, I would not expect that to be linear given that things like number of cx is not linear. However, the truncation of a full width circuit down to the subset is something that is likely linear. However, the data would suggest that that truncation is much more time consuming than the noise modeling itself (that should not depend on the system size)

**Comment by chriseclectic** [comment]
This should be somewhat fixed in 0.9 release which removes the need for truncation of the noise model. Note that there will still be some scaling when the model is converted to kraus or superop form for large numbers of qubits, since the entire model is still converted (but once, rather than truncated and converted for every circuit). This is something that could maybe be further improved upon but would require some structural changes to the C++ noise model.
